x-pack/plugins/observability_solution/observability_ai_assistant/server/rule_connector/index.ts

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { filter, lastValueFrom } from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { KibanaRequest, Logger } from '@kbn/core/server';
+import { AlertingConnectorFeatureId } from '@kbn/actions-plugin/common/connector_feature_config';
+import type {
+  ActionType as ConnectorType,
+  ActionTypeExecutorOptions as ConnectorTypeExecutorOptions,
+  ActionTypeExecutorResult as ConnectorTypeExecutorResult,
+} from '@kbn/actions-plugin/server/types';
+import { ConnectorAdapter } from '@kbn/alerting-plugin/server';
+import { ObservabilityAIAssistantRouteHandlerResources } from '../routes/types';
+import {
+  ChatCompletionChunkEvent,
+  MessageRole,
+  StreamingChatResponseEventType,
+} from '../../common';
+import { OBSERVABILITY_AI_ASSISTANT_CONNECTOR_ID } from '../../common/rule_connector';
+import { concatenateChatCompletionChunks } from '../../common/utils/concatenate_chat_completion_chunks';
+import { ChatFunctionClient } from '../service/chat_function_client';
+
+const ParamsSchema = schema.object({
+  connector: schema.string(),
+  message: schema.string({ minLength: 1 }),
+});
+
+const RuleSchema = schema.object({
+  id: schema.string(),
+  name: schema.string(),
+  tags: schema.arrayOf(schema.string(), { defaultValue: [] }),
+  ruleUrl: schema.nullable(schema.string()),
+});
+
+const AlertStateSchema = schema.oneOf([
+  schema.literal('new'),
+  schema.literal('ongoing'),
+  schema.literal('recovered'),
+  schema.literal('unknown'),
+]);
+
+const ConnectorParamsSchema = schema.object({
+  connector: schema.string(),
+  message: schema.string({ minLength: 1 }),
+  rule: RuleSchema,
+  alertState: AlertStateSchema,
+});
+
+type AlertState = TypeOf<typeof AlertStateSchema>;
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+export type ConnectorParamsType = TypeOf<typeof ConnectorParamsSchema>;
+type RuleType = TypeOf<typeof RuleSchema>;
+
+export type ObsAIAssistantConnectorType = ConnectorType<{}, {}, ConnectorParamsType, unknown>;
+
+export type ObsAIAssistantConnectorTypeExecutorOptions = ConnectorTypeExecutorOptions<
+  {},
+  {},
+  ConnectorParamsType
+>;
+
+export function getObsAIAssistantConnectorType(
+  initResources: (request: KibanaRequest) => Promise<ObservabilityAIAssistantRouteHandlerResources>
+): ObsAIAssistantConnectorType {
+  return {
+    id: OBSERVABILITY_AI_ASSISTANT_CONNECTOR_ID,
+    isSystemActionType: true,
+    getKibanaPrivileges: (params) => ['api:observabilityAIAssistant'],
```

Comment:
I am not that familiar with the assistant privileges but what do you think about adding `app:observabilityAIAssistant`?

Reply:
Fixed in https://github.com/elastic/kibana/pull/179980/commits/f99821fdf306bb8b9565f12998c655b461fbaa97